Lib/java/std_vector.i

```diff
@@ -10,38 +10,124 @@
 %}
 
 %define SWIG_STD_VECTOR_MINIMUM_INTERNAL(CTYPE, CREF_TYPE)
+%typemap(javabase) std::vector<CTYPE> "java.util.AbstractList<$typemap(jboxtype, CTYPE)>"
+%typemap(javainterfaces) std::vector<CTYPE> "java.util.RandomAccess"
+%typemap(javacode) std::vector<CTYPE> %{
+  public $javaclassname($typemap(jstype, CTYPE)[] initialElements) {
+    this();
+    for ($typemap(jstype, CTYPE) element : initialElements) {
+      add(element);
+    }
+  }
+
+  public $javaclassname(Iterable<$typemap(jboxtype, CTYPE)> initialElements) {
+    this();
+    for ($typemap(jstype, CTYPE) element : initialElements) {
+      add(element);
+    }
+  }
+
+  public $typemap(jboxtype, CTYPE) get(int index) {
+    return doGet(index);
+  }
+
+  public $typemap(jboxtype, CTYPE) set(int index, $typemap(jboxtype, CTYPE) e) {
+    return doSet(index, e);
+  }
+
+  public boolean add($typemap(jboxtype, CTYPE) e) {
+    modCount++;
+    doAdd(e);
+    return true;
+  }
+
+  public void add(int index, $typemap(jboxtype, CTYPE) e) {
+    modCount++;
+    doAdd(index, e);
+  }
+
+  public $typemap(jboxtype, CTYPE) remove(int index) {
+    modCount++;
+    return doRemove(index);
+  }
+
+  protected void removeRange(int fromIndex, int toIndex) {
+    modCount++;
+    doRemoveRange(fromIndex, toIndex);
+  }
+%}
+
     public:
         typedef size_t size_type;
         typedef CTYPE value_type;
         typedef CREF_TYPE const_reference;
         vector();
         vector(size_type n);
-        size_type size() const;
+        // Note that it returns int, and not size_type, for compatibility with
+        // the signature of size() in Java interface.
+        int size() const;
```

Comment:
Hmm, I initially decided to use `int` for `size()` return value because it seemed that Java couldn't support vectors with more than `2^31` elements anyhow, but I've just found #674 and it's probably a good idea to throw an exception if the C++ vector is larger than that as that PR does...